bridges/EBayBridge.php

```diff
@@ -61,56 +71,107 @@ public function collectData()
-            }
+            // Define a closure to shorten the ugliness of querying the current listing.
+            $find = function ($query, $altText = '') use ($listing) {
+                return $listing->find($query, 0)->plaintext ?? $altText;
+            };
 
-            $listingTitle = $listing->find('.s-item__title', 0);
-            if ($listingTitle) {
-                $item['title'] = $listingTitle->plaintext;
+            $item['title'] = $find('.s-item__title');
+            if (!$item['title']) {
+                // Skip entries where the title cannot be found (for w/e reason).
+                continue;
             }
 
-            $subtitle = implode('', $listing->find('.s-item__subtitle'));
-
-            $listingUrl = $listing->find('.s-item__link', 0);
-            if ($listingUrl) {
-                $item['uri'] = $listingUrl->href;
+            // It appears there may be more than a single 'subtitle' subclass in the listing. Collate them.
+            $subtitles = $listing->find('.s-item__subtitle');
+            if (is_array($subtitles)) {
+                $subtitle = trim(implode(' ', array_map(function ($s) { return $s->plaintext; }, $subtitles)));
             } else {
-                $item['uri'] = null;
+                $subtitle = trim($subtitles->plaintext ?? '');
             }
 
+            // Get the listing's link and uid.
+            $item['uri'] = $listing->find('.s-item__link', 0)?->href ?? null;
             if (preg_match('/.*\/itm\/(\d+).*/i', $item['uri'], $matches)) {
                 $item['uid'] = $matches[1];
             }
 
-            $priceDom = $listing->find('.s-item__details > .s-item__detail > .s-item__price', 0);
-            $price = $priceDom->plaintext ?? 'N/A';
+            // Price should be fetched on its own so we can provide the alt text without complication.
+            $price = $find('.s-item__price', '[NO PRICE]');
+
+            // Map a list of dynamic variable names to their subclasses within the listing.
+            //   This is just a bit of sugar to make this cleaner and more maintainable.
+            $propertyMappings = [
+                'additionalPrice'   => '.s-item__additional-price',
+                'discount'          => '.s-item__discount',
+                'shippingFree'      => '.s-item__freeXDays',
+                'localDelivery'     => '.s-item__localDelivery',
+                'logisticsCost'     => '.s-item__logisticsCost',
+                'location'          => '.s-item__location',
+                'obo'               => '.s-item__formatBestOfferEnabled',
+                'sellerInfo'        => '.s-item__seller-info-text',
+                'bids'              => '.s-item__bidCount',
+                'timeLeft'          => '.s-item__time-left',
+                'timeEnd'           => '.s-item__time-end',
+            ];
+
+            foreach ($propertyMappings as $k => $v) {
+                $$k = $find($v);
```

Comment:
variable variable not allowed

Reply:
well maybe we can make an exception.

it's been years since i seen variable variable in php. and i remember it being a security risk. but i think a hardcoded list is okay